weave

```diff
@@ -328,15 +300,82 @@ fi
 # main (remote and --local) - settings
 ######################################################################
 
+## Versions ##
+SCRIPT_VERSION="unreleased"
+IMAGE_VERSION=latest
+[ "$SCRIPT_VERSION" = "unreleased" ] || IMAGE_VERSION=$SCRIPT_VERSION
+IMAGE_VERSION=${WEAVE_VERSION:-$IMAGE_VERSION}
+
+## Image Names ##
+
+DOCKERHUB_USER=${DOCKERHUB_USER:-weaveworks}
+
+### Detect the architecture this script is running on
+case "$(uname -m)" in
+  x86_64*)
+    ARCH=amd64;;
+  i?86_64*)
+    ARCH=amd64;;
+  amd64*)
+    ARCH=amd64;;
+  aarch64*)
+    ARCH=arm64;;
+  arm64*)
+    ARCH=arm64;;
+  arm*)
+    ARCH=arm;;
+  *)
+    echo "Unsupported host architecture. Must be amd64, arm or arm64."
+    exit 1;;
+esac
+
+# Temporary workaround until docker 1.10+, which supports manifest lists, is supported by Weave
+if [[ ${ARCH} == "amd64" ]]; then
+    # amd64 does not use the suffix
+    ARCH_IMAGE_EXT=""
+else
+    # arm and arm64 has -arch-suffixed images
+    ARCH_IMAGE_EXT="-${ARCH}"
+fi
+
+# These are needed for remote execs, hence we introduce them here
+BASE_EXEC_IMAGE=${DOCKERHUB_USER}/weaveexec${ARCH_IMAGE_EXT}
+EXEC_IMAGE=${BASE_EXEC_IMAGE}:${IMAGE_VERSION}
```

Comment:
You cannot move this after the first use (which is in the call to exec_remote on line 260)

I'm going to stop checking these now; please leave things in the same order or justify the movement.